django/db/models/base.py

```diff
@@ -369,6 +370,15 @@ def _default_manager(cls):
         return cls._meta.default_manager
 
 
+class ModelBase(AltersDataMixin, ModelMetaclass):
+    """
+    We can't have ModelMetaclass inherit from AltersDataMixin.
+    ModelMetaclass does not pass all the attrs to it's super class.
+    So, AltersDataMixin will not be able set alters_data attribute
+    on the updated definition of subclass methods
+    """
```

Comment:
method definitions are not available as part of `attrs`. So `add_to_class` doesn't help.
I've refactored the logic of setting `alters_data` attribute into util function and calling it inside existing meta classes.